Generals/Code/GameEngine/Include/GameLogic/Object.h

```diff
@@ -420,6 +420,7 @@ class Object : public Thing, public Snapshot
 	void friend_setContainedBy( Object *containedBy ) { m_containedBy = containedBy; }
 	const Object* getEnclosingContainedBy() const; ///< Find the first enclosing container in the containment chain.
 	const Object* getOuterObject() const; ///< Get the top-level object
+	Bool isLogicallyVisible() const; ///< Returns whether the object is logically visible to the player.
```

Comment:
Maybe move this function somewhere more relatable, because it does not belong to the contained-by functions bunch. Maybe move near `isLocallyViewed`.